mempool/mempool.go

```diff
@@ -517,8 +517,9 @@ func (mem *Mempool) ReapMaxTxs(max int) types.Txs {
 		time.Sleep(time.Millisecond * 10)
 	}
 
-	txs := make([]types.Tx, 0, cmn.MinInt(mem.txs.Len(), max))
-	for e := mem.txs.Front(); e != nil && len(txs) <= max; e = e.Next() {
```

Comment:
note: `e != nil` prevents us from going over mem.txs.Len()

Reply:
I know，I think "len(txs) <= max" is not precise, and It's a little redundant to judge twice for "e != nil && len(txs) <= max",
if we modify it to numTxsToReap , "e != nil" is without need, what do you think of

Reply:
Agree, if we switch to `numTxsToReap`, we can remove `e != nil`

Reply:
thanks for review :)

Reply:
note for myself: need to check if mem.txs can be mutated while we iterate over it in ReapMaxTxs